### PR TITLE
Implemented reading project id from an environment variable

### PR DESCRIPTION
**Pull request by raghav0110**
Adding the ability to read the Gcloud Pubsub project id from an environment variable.
Updated the README and gitignore.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -10,6 +10,9 @@ The current version of the proxy utilizes Google Cloud Pub/Sub as the backend Pu
 If you're planning on running multiple instances of the proxy for scalability,
 you must avoid running multiple instances on the same host.
 
+If you will be using Google Cloud Pub/Sub, you must set the GCLOUD_PUBSUB_MQTT_PROXY environment
```
Comment (tmatsuo) [inline README.md]: I would use `PROJECT_ID` or `CPS_MQTT_PROXY_PROJECT_ID`.

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 2]
```diff
@@ -96,7 +96,7 @@
   private static final String PREFIX = "cps-";
   private static final String ASTERISK_URLENCODE_VALUE = "%2A";
   // TODO - read project name from config file
-  private static final String CLOUD_PUBSUB_PROJECT_NAME = "iot-cloud-pubsub";
+  private static final String CLOUD_PUBSUB_PROJECT_NAME = System.getenv("GCLOUD_PUBSUB_MQTT_PROXY");
```
Comment (tmatsuo) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: What happens if there's no such environment variable?

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 3]
```diff
@@ -95,8 +95,7 @@
   private static final String HASH_PREFIX = "sha256-";
   private static final String PREFIX = "cps-";
   private static final String ASTERISK_URLENCODE_VALUE = "%2A";
-  // TODO - read project name from config file
-  private static final String CLOUD_PUBSUB_PROJECT_NAME = "iot-cloud-pubsub";
+  private static final String CLOUD_PUBSUB_PROJECT_NAME = System.getenv("GCLOUD_PUBSUB_PROJECT_ID");
```
Comment (tmatsuo) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: So what happens if there's no such environment variable? If the error message is user friendly enough, I'm fine, but if it's mysterious, I'd rather provide an error message saying "Please set the GCLOUD_PUBSUB_PROJECT_ID environment variable to run this program".

Comment (raghav0110) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: Done.

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 4]
```diff
@@ -122,6 +121,9 @@
    * @throws IOException when the initialization of the Google Cloud Pub/Sub client fails.
    */
   public GcloudPubsub() throws IOException {
+    if (CLOUD_PUBSUB_PROJECT_NAME == null) {
+      throw new IllegalStateException("Please set the GCLOUD_PUBSUB_PROJECT_ID environment variable to your project id");
```
Comment (tmatsuo) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: Can the error message and the name of the environment variable be constants?

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 5]
```diff
@@ -146,6 +148,9 @@ public GcloudPubsub() throws IOException {
    * @param pubsub the Google Cloud Pub/Sub instance to use for Pub/Sub operations.
    */
   public GcloudPubsub(Pubsub pubsub) {
+    if (CLOUD_PUBSUB_PROJECT_NAME == null) {
+      throw new IllegalStateException("Please set the GCLOUD_PUBSUB_PROJECT_ID environment variable to your project id");
```
Comment (tmatsuo) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: ditto

File: README.md [hunk 6]
```diff
@@ -10,6 +10,9 @@ The current version of the proxy utilizes Google Cloud Pub/Sub as the backend Pu
 If you're planning on running multiple instances of the proxy for scalability,
 you must avoid running multiple instances on the same host.
 
+If you will be using Google Cloud Pub/Sub, you must set the GCLOUD_PUBSUB_PROJECT_ID environment
```
Comment (aozarov) [inline README.md]: In gcloud (Veneer) libraries we decided to use GOOGLE_CLOUD_PROJECT.
see: https://github.com/GoogleCloudPlatform/gcloud-meta/issues/17

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java (hunk 3 above)
Comment (aozarov) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: If you want you can copy/paste some code from https://github.com/GoogleCloudPlatform/gcloud-java/blob/master/gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java#L194 that tries to auto-detect the project id.

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 7]
```diff
@@ -95,12 +95,16 @@
   private static final String HASH_PREFIX = "sha256-";
   private static final String PREFIX = "cps-";
   private static final String ASTERISK_URLENCODE_VALUE = "%2A";
-  // TODO - read project name from config file
-  private static final String CLOUD_PUBSUB_PROJECT_NAME = "iot-cloud-pubsub";
+  private static final String GCLOUD_PUBSUB_PROJECT_ID_ENV_VARIABLE = "GCLOUD_PUBSUB_PROJECT_ID";
+  private static final String CLOUD_PUBSUB_PROJECT_ID =
+      System.getenv(GCLOUD_PUBSUB_PROJECT_ID_ENV_VARIABLE);
+  private static final String GCLOUD_PUBSUB_PROJECT_ID_NOT_SET_ERROR =
+      "Please set the" + GCLOUD_PUBSUB_PROJECT_ID_ENV_VARIABLE
```
Comment (aozarov) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: the " (missing space) and also in " environment"

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java [hunk 8]
```diff
@@ -95,12 +95,16 @@
...
+      + "environment variable to be equal to your project id";
```
Comment (aozarov) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: I prefer: "Please set the " + GCLOUD_PUBSUB_PROJECT_ID_ENV_VARIABLE + " environment variable to your project id" (without the "be equal")

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java (hunk 7 above)
Comment (raghav0110) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: Done.

File: mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java (hunk 8 above)
Comment (raghav0110) [inline mqtt_cloud_pubsub_proxy/src/main/java/com/google/cloud/pubsub/proxy/gcloud/GcloudPubsub.java]: Done.

File: README.md (hunk 6 above)
Comment (raghav0110) [inline README.md]: I added a TODO for being consistent with the gcloud libs.

Comment (tmatsuo) [inline README.md]: Why not changing it now? It is confusing if we change the environment variable name.

Comment (raghav0110) [inline README.md]: Done.

**Comment by tmatsuo** [comment]
LGTM

**Pull request merged**